src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -666,16 +666,20 @@ export class Response extends AbstractResponse implements IDisposable {
 			}
 			this._updateRepr(quiet);
 		} else if (progress.kind === 'textEdit' || progress.kind === 'notebookEdit') {
-			// If the progress.uri is a cell Uri, its possible its part of the inline chat.
-			// Old approach of notebook inline chat would not start and end with notebook Uri, so we need to check for old approach.
-			const useOldApproachForInlineNotebook = progress.uri.scheme === Schemas.vscodeNotebookCell && !this._responseParts.find(part => part.kind === 'notebookEditGroup');
 			// merge edits for the same file no matter when they come in
-			const notebookUri = useOldApproachForInlineNotebook ? undefined : CellUri.parse(progress.uri)?.notebook;
+			const notebookUri = CellUri.parse(progress.uri)?.notebook;
 			const uri = notebookUri ?? progress.uri;
 			let found = false;
 			const groupKind = progress.kind === 'textEdit' && !notebookUri ? 'textEditGroup' : 'notebookEditGroup';
 			// eslint-disable-next-line @typescript-eslint/no-explicit-any
-			const edits: any = groupKind === 'textEditGroup' ? progress.edits : progress.edits.map(edit => TextEdit.isTextEdit(edit) ? { uri: progress.uri, edit } : edit);
+			const edits: any = // TextEdit[] | (ICellTextEditOperation | ICellEditOperation)[] =
```

Comment:
this is a typing puzzle that I wasn't able to solve. I don't think it likes the mixture of cell / celltext edit operations. Co-pilot's solution was to triplicate this section of code and still involved an `as` cast.

I think we're forcing too many operation types through the same method here.

Reply:
Yea I think we should just split this out, or maybe have a helper function with some generic parts that lets us express things correctly. THis specific `any` cast was already a source of a bug in the past